### PR TITLE
Added support for more bone weights in GLTF2 

### PR DESCRIPTION
**Pull request by Promit**
The GLTF2 importer doesn't actually read beyond the first four bone weights (first attribute). This patch expands the parser to read as many bone weights as are available in the file.

### CODE REVIEW

File: code/AssetLib/glTF2/glTF2Importer.cpp [hunk 1]
```diff
@@ -967,21 +967,29 @@ static void BuildVertexWeightMapping(Mesh::Primitive &primitive, std::vector<std
     struct Weights {
         float values[4];
     };
-    Weights *weights = nullptr;
-    attr.weight[0]->ExtractData(weights);
+    Weights **weights = new Weights*[attr.weight.size()];
```
[the remark this answers is not in the document]
Comment (Promit) [reply code/AssetLib/glTF2/glTF2Importer.cpp:970]: I agree, I'm going to rewrite it with vectors.

[the remark this answers is not in the document]
Comment (kimkulling) [reply code/AssetLib/glTF2/glTF2Importer.cpp:970]: Any progress regarding your approach using std::vector?

File: code/AssetLib/glTF2/glTF2Importer.cpp [hunk 2]
```diff
@@ -990,17 +998,26 @@ static void BuildVertexWeightMapping(Mesh::Primitive &primitive, std::vector<std
...
-            if (weight > 0 && bone < map.size()) {
-                map[bone].reserve(8);
-                map[bone].emplace_back(static_cast<unsigned int>(i), weight);
+    for (size_t w = 0; w < attr.weight.size(); ++w) {
+        for (size_t i = 0; i < num_vertices; ++i) {
+            for (int j = 0; j < 4; ++j) {
+                const unsigned int bone = (indices8 != nullptr) ? indices8[w][i].values[j] : indices16[w][i].values[j];
+                const float weight = weights[w][i].values[j];
+                if (weight > 0 && bone < map.size()) {
+                    map[bone].reserve(8);
```
Comment (kimkulling) [inline code/AssetLib/glTF2/glTF2Importer.cpp:1007]: Use of magic number.

**Review by kimkulling — CHANGES_REQUESTED** [review]

File: code/AssetLib/glTF2/glTF2Importer.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kimkulling) [reply code/AssetLib/glTF2/glTF2Importer.cpp:973]: I guess I will take care.

**Review by kimkulling — APPROVED** [review]

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**